sonic-chassisd/scripts/chassisd

```diff
@@ -559,6 +612,161 @@ class ModuleUpdater(logger.Logger):
                         self._cleanup_chassis_app_db(module)
                     self.down_modules[module]['cleaned'] = True
 
+#
+# Module Updater ==============================================================
+#
+
+
+class SmartSwitchModuleUpdater(ModuleUpdater):
+
+    def __init__(self, log_identifier, chassis, my_slot, supervisor_slot):
+        """
+        Constructor for ModuleUpdater
+        :param chassis: Object representing a platform chassis
+        """
+        super(ModuleUpdater, self).__init__(log_identifier)
+
+        self.chassis = chassis
+        self.my_slot = my_slot
+        self.supervisor_slot = supervisor_slot
+        self.num_modules = self.chassis.get_num_modules()
+        # Connect to STATE_DB and create chassis info tables
+        state_db = daemon_base.db_connect("STATE_DB")
+        self.chassis_table = swsscommon.Table(state_db, CHASSIS_INFO_TABLE)
+        self.module_table = swsscommon.Table(state_db, CHASSIS_MODULE_INFO_TABLE)
+        self.midplane_table = swsscommon.Table(state_db, CHASSIS_MIDPLANE_INFO_TABLE)
+        self.info_dict_keys = [CHASSIS_MODULE_INFO_NAME_FIELD,
+                               CHASSIS_MODULE_INFO_DESC_FIELD,
+                               CHASSIS_MODULE_INFO_SLOT_FIELD,
+                               CHASSIS_MODULE_INFO_SERIAL_FIELD,
+                               CHASSIS_MODULE_INFO_OPERSTATUS_FIELD]
+
+        self.chassis_state_db = daemon_base.db_connect("CHASSIS_STATE_DB")
+
+        self.hostname_table = swsscommon.Table(self.chassis_state_db, CHASSIS_MODULE_HOSTNAME_TABLE)
+        self.module_reboot_table = swsscommon.Table(self.chassis_state_db, CHASSIS_MODULE_REBOOT_INFO_TABLE)
+        self.down_modules = {}
+        self.chassis_app_db_clean_sha = None
+
+        self.linecard_reboot_timeout = DEFAULT_LINECARD_REBOOT_TIMEOUT
+        if os.path.isfile(PLATFORM_ENV_CONF_FILE):
+            with open(PLATFORM_ENV_CONF_FILE, 'r') as file:
+                for line in file:
+                    field = line.split('=')[0].strip()
+                    if field == "linecard_reboot_timeout":
+                        self.linecard_reboot_timeout = int(line.split('=')[1].strip())
+
```

Comment:
@rameshraghupathy line card reboot .? I am not aware of this in smart switch HLD

Reply:
Good catch!  Cleaned it!
@pr